pkg/configuration/configuration.go

```diff
@@ -310,6 +313,9 @@ func (c *Registry) GetVerificationMessageTemplate() string {
 // is not required
 func (c *Registry) GetVerificationExcludedEmailDomains() []string {
 	if c.excludedDomains == nil {
+		lock.Lock()
```

Comment:
Unfortunately it's not enough. We can't read and write the same variable in two different goroutines in the same time. And with this approach it's possible for one goroutine to check `if c.excludedDomains == nil` while the other one is doing `c.excludedDomains = strings.Split(c.v.GetString(varVerificationExcludedEmailDomains), ",")`
We should use RWMutex instead. But I think it will be simpler just to initialize `c.excludeDomains` in `New()` or in `CreateEmptyRegistry()`

Reply:
I've put the lock() around the whole function and modified the signup service (which uses the excluded domains value) so that it will only read it once.

Reply:
While it's probably not super critical in this particular case it's still a bad pattern to lock the entire function. We are basically making the entire verification service a single threaded. Let's avoid that.

Reply:
Ah.. I think I got what you mean. You are modifying the configuration usage too. So, the service won't read the configuration every time. It solves the problem with multi-threading but error prone IMO. We will need to remember that we are not supposed to read this particular configuration as we do for the other configuration. Why not just initiate this configuration once in the constructor?

Reply:
Code moved to constructor.